### ISSUE TITLE
 onnx-export-TensorRT  Error

### ISSUE THREAD

**Issue by J1T2** [roboflow/rf-detr#1228]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

Pth trained by my RFDETRSegNano is exported to onnx normally, and seg results can be inferred normally. However, there is an error in exporting from onnx to TensorRT, which is beyond my ability to solve the problem. Do you have any partners who have solved it? Please provide help and thank you!
The following is the error content:
```
[07/23/2026-09:54:56] [E] [TRT] ModelImporter.cpp:771: While parsing node number 519 [TopK -> "/transformer/TopK_output_0"]:
[07/23/2026-09:54:56] [E] [TRT] ModelImporter.cpp:772: --- Begin node ---
[07/23/2026-09:54:56] [E] [TRT] ModelImporter.cpp:773: input: "/transformer/ReduceMax_output_0"
input: "/transformer/Constant_38_output_0"
output: "/transformer/TopK_output_0"
output: "/transformer/TopK_output_1"
name: "/transformer/TopK"
op_type: "TopK"
attribute {
  name: "axis"
  i: 1
  type: INT
}
attribute {
  name: "largest"
  i: 1
  type: INT
}
attribute {
  name: "sorted"
  i: 1
  type: INT
}
doc_string: "D:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\rfdetr\\models\\transformer.py(355): forward\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1769): _slow_forward\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1790): _call_impl\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1779): _wrapped_call_impl\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\rfdetr\\models\\lwdetr.py(609): forward_export\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1769): _slow_forward\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1790): _call_impl\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1779): _wrapped_call_impl\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\jit\\_trace.py(131): wrapper\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\jit\\_trace.py(140): forward\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1790): _call_impl\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\nn\\modules\\module.py(1779): _wrapped_call_impl\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\jit\\_trace.py(1439): _get_trace_graph\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\onnx\\_internal\\torchscript_exporter\\utils.py(903): _trace_and_get_graph_from_model\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\onnx\\_internal\\torchscript_exporter\\utils.py(997): _create_jit_graph\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\onnx\\_internal\\torchscript_exporter\\utils.py(1113): _model_to_graph\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\onnx\\_internal\\torchscript_exporter\\utils.py(1515): _export\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\onnx\\_internal\\torchscript_exporter\\utils.py(552): export\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\torch\\onnx\\__init__.py(334): export\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\rfdetr\\export\\_onnx\\exporter.py(107): export_onnx\nD:\\SoftWare\\miniconda3\\envs\\rfdetr\\lib\\site-packages\\rfdetr\\detr.py(1284): export\ne:\\XKZZ\\XK-Project\\python\\rf-detr-develop\\RF-DETR\\pytorch_export_onnx.py(10): <module>\n"

[07/23/2026-09:54:56] [E] [TRT] ModelImporter.cpp:774: --- End node ---
[07/23/2026-09:54:56] [E] [TRT] ModelImporter.cpp:777: ERROR: onnx2trt_utils.cpp:342 In function convertAxis:
[8] Assertion failed: (axis >= 0 && axis <= nbDims) && "Axis must be in the range [0, nbDims]."
[07/23/2026-09:54:56] [E] Failed to parse onnx file
[07/23/2026-09:54:56] [I] Finished parsing network model. Parse time: 0.273253
[07/23/2026-09:54:56] [E] Parsing model failed
[07/23/2026-09:54:56] [E] Failed to create engine from model or file.
[07/23/2026-09:54:56] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8601] # trtexec.exe --onnx=rfdetr-seg-nano-sim.onnx --saveEngine=rfdetr-seg-nano-sim.engine --fp16
```
详细解释报错

### Environment

rfdetr                             1.8.3 
torch                              2.11.0+cu128
torchaudio                     2.11.0+cu128
tensorrt                          8.6.1..6

### Minimal Reproducible Example

trtexec.exe --onnx=rfdetr-seg-nano.onnx --saveEngine=rfdetr-seg-nano.engine --fp16

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by Nishant-ZFYII** [comment]
Good Day, 

I would like to work on this bug. 

I exported `RFDETRSegNano` (both current `develop` and the exact `rfdetr==1.8.3` you're using) to ONNX and built a TensorRT `8.6.1` FP16 engine from it on Linux matching your TensorRT patch version with both a static `batch=1` export and a dynamic-batch export. Both parsing and engine building completed successfully every time; I couldn't reproduce the `TopK` axis error on the stock pretrained checkpoint.

A few details would help us narrow this down:

1. Is the model you're exporting the **stock pretrained checkpoint**, or a **fine-tuned checkpoint** on your own dataset? If fine-tuned, what `num_classes` did you train with? A different class count reshapes the exact tensor feeding into the failing `TopK`.
2. What `--shape` (if any) did you pass to `model.export()`? Or did you use the default resolution?
3. Did you run any ONNX post-processing (`onnxsim`, `polygraphy surgeon`, graph simplification, etc.) between exporting and running `trtexec`, or is `trtexec` running directly against the raw `inference_model.onnx`?
4. Since you're on `trtexec.exe`, can you confirm this is **Windows**? TensorRT ships platform-specific builds, and that's my leading suspect for why I can't reproduce on Linux with the same TensorRT version.

If you're able to share the `.onnx` file (or just the model config / a minimal repro checkpoint), that would help me to pin this down.


Thanks and Happy to Help :)

**Comment by Borda** [comment]
@Nishant-ZFYII could you pls validate if this is still an issue after the refactor in #853?

**Comment by Borda** [comment]
I believe this was resolved in #853 and #1242 🐰 please try the latest release 1.9.0+
However, if you still observe these issues, pls feel free to reopen it 🦝
